langchain/docstore/base.py

```diff
@@ -15,3 +15,9 @@ def search(self, search: str) -> Union[str, Document]:
         If page exists, return the page summary, and a Document object.
         If page does not exist, return similar entries.
         """
+
+
+class AddableMixin(ABC):
+    @abstractmethod
+    def add(self, texts: Dict[str, Document]) -> None:
```

Comment:
type signature of text changes from `Dict` to `List`, would having a unique identifier as an attribute of the document model help make the api more uniform?

Reply:
we will need the ids... some potential options:

1. List[Document] and then have this return list of ids (eg dont pass ids in but return them, i think that would work)
2. List[Tuple[str, Document]] (list of ids and documents
3. keep as is

any strong preferences? at moment i dont have one, not enough examples for me to be confident in this abstraction, so honestly im with whatever for now acknowledging it may change, but maybe one of you has done something similar before and so knows where this is likely going?